adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala

```diff
@@ -202,6 +202,72 @@ class AlignmentRecordConverter extends Serializable {
 
     samHeader
   }
+
+  /**
+   * Converts a fragment to a set of reads.
+   *
+   * @param fragment Fragment to convert.
+   * @return The collection of alignments described by the fragment. If the fragment
+   *         doesn't contain any alignments, this method will return one unaligned
+   *         AlignmentRecord per sequence in the Fragment.
+   */
+  def convertFragment(fragment: Fragment): Iterable[AlignmentRecord] = {
+    val sequences: List[Sequence] = asScalaBuffer(fragment.getSequences).toList
+    require(sequences.size > 0, "Fragment %s contains no sequences.".format(fragment))
+
+    // create fragment builders
+    val builders = if (fragment.getAlignments.length == 0) {
+      var id = 0
```

Comment:
Actually, that won't work, this will...

```
if (fragment.getAlignments.isEmpty) {
      List.tabulate(sequences.length)(id =>
        AlignmentRecord.newBuilder()
          .setSequence(sequences(id).getBases)
          .setQual(sequences(id).getQualities)
          .setReadNum(id))
    } else {
```